image/bmp.ksy

```diff
@@ -17,31 +17,104 @@ meta:
+
+  ## Samples
+
+  Great collection of various BMP sample files:
+  [**BMP Suite Image List**](
+    http://entropymine.com/jason/bmpsuite/bmpsuite/html/bmpsuite.html
+  ) (by Jason Summers)
+
+  If only there was such a comprehensive sample suite for every file format! It's like
+  a dream for every developer of any binary file format parser. It contains a lot of
+  different types and variations of BMP files, even the tricky ones, where it's not clear
+  from the specification how to deal with them (marked there as "**q**uestionable").
+
+  If you make a program which will be able to read all the "**g**ood" and "**q**uestionable"
+  BMP files and won't crash on the "**b**ad" ones, it will definitely have one of the most
+  extensive support of BMP files in the universe!
+
+  ## BITMAPV2INFOHEADER and BITMAPV3INFOHEADER
+
+  A beneficial discussion on Adobe forum (archived):
+  [**Invalid BMP Format with Alpha channel**](
+    https://web.archive.org/web/20150127132443/https://forums.adobe.com/message/3272950
+  )
+
+  In 2010, someone noticed that Photoshop generated BMP with an odd type of header. There wasn't
+  any documentation available for this header at the time (and still isn't).
+  However, Chris Cox (former Adobe employee) claimed that they hadn't invented any type
+  of proprietary header and everything they were writing was taken directly
+  from the Microsoft documentation.
+
+  It showed up that the unknown header was called BITMAPV3INFOHEADER.
+  Although Microsoft has apparently requested and verified the use of the header,
+  the documentation on MSDN has probably got lost and they have probably
+  forgotten about this type of header.
+
+  This is the only source I could find about these structures, so we could't rely
+  on it so much, but I think supporting them as a read-only format won't harm anything.
+  Due to the fact that it isn't documented anywhere else, most applications don't support it.
+
+  All Windows headers at once (including mentioned BITMAPV2INFOHEADER and BITMAPV3INFOHEADER):
+  ![Bitmap headers overview](
+    https://forums.adobe.com/servlet/JiveServlet/showImage/2-3273299-47801/BMP_Headers.png
+  )
+
+  ## Specs
+   * [Bitmap Storage (Windows Dev Center)](
+       https://docs.microsoft.com/en-us/windows/win32/gdi/bitmap-storage
+     )
+      * BITMAPFILEHEADER
+      * BITMAPINFOHEADER
+      * BITMAPV4HEADER
+      * BITMAPV5HEADER
+   * [OS/2 Bitmap File Format](
+        https://www.fileformat.info/format/os2bmp/egff.htm
+     )
+      * BITMAPFILEHEADER (OS2BMPFILEHEADER)
+      * BITMAPCOREHEADER (OS21XBITMAPHEADER)
+      * OS22XBITMAPHEADER
+   * [Microsoft Windows Bitmap](
+        http://netghost.narod.ru/gff/graphics/summary/micbmp.htm
+     )
+      * BITMAPFILEHEADER (WINBMPFILEHEADER)
+      * BITMAPCOREHEADER (WIN2XBITMAPHEADER)
+      * BITMAPINFOHEADER (WINNTBITMAPHEADER)
+      * BITMAPV4HEADER (WIN4XBITMAPHEADER)
+
 seq:
-  - id: file_hdr
+  - id: file_header
     type: file_header
```

Comment:
Technically, this [breaks C# code generation](https://github.com/kaitai-io/kaitai_struct/issues/86), but that's fine — it's still a code generation problem, not a format problem.

Reply:
I renamed it back to `file_hdr`, it might not be a format problem, but it's anyway probably a good thing to have distinct names for type and field because one can instantly see what belongs to what.